realm/realm-library/src/androidTest/java/io/realm/RealmListTests.java

```diff
@@ -231,18 +219,16 @@ public void testSet_managedMode() {
     }
 
     @Test
-    public void testSetNull_nonManagedMode() {
+    public void set_nullInNonManagedMode() {
+        @SuppressWarnings("MismatchedQueryAndUpdateOfCollection")
         RealmList<AllTypes> list = new RealmList<AllTypes>();
         list.add(new AllTypes());
-        try {
-            list.set(0, null);
-            fail("Setting a null value should result in a exception");
-        } catch (IllegalArgumentException ignore) {
-        }
+        thrown.expect(IllegalArgumentException.class);
```

Comment:
I am only using `@Test(expected` for one line testing, to prevent manual mistakes..